trestle/core/commands/merge.py

```diff
@@ -14,119 +14,131 @@
-        current_model: Type[OscalBaseModel],
-        current_filename: str,
-        initial_path: Path = None,
-        visited_elements: Set[str] = None
-    ):
-        """List paths that can be used in the -e option for the merge operation."""
-        if initial_path is None:
-            initial_path = cwd
-        if visited_elements is None:
-            visited_elements = set()
-
-        path_sep = '.' if current_alias else ''
-
-        # List options for merge
-        if not utils.is_collection_field_type(current_model):
-
-            malias = current_alias.split('.')[-1]
-            if cwd.is_dir() and malias != fs.extract_alias(cwd):
-                split_subdir = cwd / malias
-            else:
-                split_subdir = cwd.parent / cwd.with_suffix('').name
-
-            # Go through each file or subdirectory in the cwd
-            fields_by_alias = current_model.alias_to_field_map()
-            for filepath in Path.iterdir(split_subdir):
-                if filepath.is_file() and cwd == initial_path:
-                    continue
-
-                alias = filepath.with_suffix('').name
-                if alias in fields_by_alias:
-                    visited_element = f'{current_alias}{path_sep}{alias}'
-                    if visited_element not in visited_elements:
-                        visited_elements.add(visited_element)
-                        self.out(f"{visited_element} (merges \'{filepath.name}\' into \'{cwd / current_filename}\')")
-
-                    # If it is subdirectory, call this function recursively
-                    if Path.is_dir(filepath):
-                        self._list_options_for_merge(
-                            filepath,
-                            f'{current_alias}{path_sep}{alias}',
-                            fields_by_alias[alias].outer_type_,
-                            f'{alias}.json',
-                            initial_path=initial_path,
-                            visited_elements=visited_elements
-                        )
+        try:
+            # FIXME: Handle multiple element paths: element_paths = args.element.split(',')
+            plan = self.merge(ElementPath(args.element))
+
+            plan.simulate()
+            plan.execute()
+        except BaseException as err:
+            logger.error(f'Merge failed: {err}')
+            return 1
+        return 0
+
+    @classmethod
+    def merge(cls, element_path: ElementPath) -> Plan:
+        """Merge operations.
+
+        It returns a plan for the operation
+        """
+        element_path_list = element_path.get_full_path_parts()
+        target_model_alias = element_path_list[-1]
+        """1. Load desination model into a stripped model"""
+        # Load destination model
+        destination_model_alias = element_path_list[-2]
+        # Destination model filename
+        destination_model_filename = Path(f'{utils.classname_to_alias(destination_model_alias, "json")}.json')
```

Comment:
This has assumed json. You should infer based on the other files which are being loaded.

Reply:
or the file path provided with -f

Reply:
I have opened an issue for it. Let me handle them when I handle the issue.